ui/widget.js

```diff
@@ -390,6 +408,55 @@ $.Widget.prototype = {
 		return this._setOptions({ disabled: true });
 	},
 
+	_classes: function( element, keys, extra, add, object ) {
+		var full = [],
+			that = this;
+
+		object = object || this.options.classes;
+
+		function processClassString( classes, checkOption ) {
+			var current, i;
+			for ( i = 0; i < classes.length; i++ ) {
+				current = that.classObject[ classes[ i ] ] ? that.classObject[ classes[ i ] ] : $();
+				that.classObject[ classes[ i ] ] = current[ add ? "add" : "not" ]( element );
```

Comment:
jQuery's `add` method order elements by DOM position, even when detached. I don't know the exact complexity of that method, but its very slow for more than a handful of elements. Is there a way to avoid using `add` completely?

If not, what are the odds for `current` containing more than 1, 10 and 100 elements?